langchain/src/llms/base.ts

```diff
@@ -63,9 +65,117 @@ export abstract class BaseLLM extends BaseLanguageModel {
     }
   }
 
+  async invoke(
+    input: BaseLanguageModelInput,
+    options?: CallOptions
+  ): Promise<string> {
+    const promptValue = BaseLLM._convertInputToPromptValue(input);
+    const result = await this.generatePrompt(
+      [promptValue],
+      options,
+      options?.callbacks
+    );
+    return result.generations[0][0].text;
+  }
+
+  // eslint-disable-next-line require-yield
+  async *_streamResponseChunks(
+    _input: string,
+    _options: this["ParsedCallOptions"],
+    _runManager?: CallbackManagerForLLMRun
+  ): AsyncGenerator<GenerationChunk> {
+    throw new Error("Not implemented.");
+  }
+
+  async *_streamBytes(
+    input: BaseLanguageModelInput,
+    options?: CallOptions
+  ): AsyncGenerator<string> {
+    yield* this._stream(input, options);
+  }
+
+  async *_stream(
+    input: BaseLanguageModelInput,
+    options?: CallOptions
+  ): AsyncGenerator<string> {
+    // Subclass check required to avoid double callbacks with default implementation
+    if (
+      this._streamResponseChunks === BaseLLM.prototype._streamResponseChunks
+    ) {
+      yield this.invoke(input, options);
+    } else {
+      const prompt = BaseLLM._convertInputToPromptValue(input);
+      const callbackManager_ = await CallbackManager.configure(
+        options?.callbacks,
+        this.callbacks,
+        options?.tags,
+        this.tags,
+        options?.metadata,
+        this.metadata,
+        { verbose: this.verbose }
+      );
+      let parsedOptions: CallOptions;
+      if (options?.timeout && !options.signal) {
+        parsedOptions = {
+          ...options,
+          signal: AbortSignal.timeout(options.timeout),
+        };
+      } else {
+        parsedOptions = (options ?? {}) as CallOptions;
+      }
+      delete parsedOptions.tags;
```

Comment:
we probably want to move this logic to a helper method?

Reply:
Ah right, will do!